### ISSUE TITLE
[BUG][EHL] i915 ERROR Suspending crtc's failed with -22

### ISSUE THREAD

**Issue by fredoh9** [thesofproject/linux#2778]
*Description
Found suspend/resume failure on `EHL_RVP_I2S`. All suspend/resume tests are failed with same error.
First time the problem is found in Daily test ID 2418(test with linux upstream PR 2772). As of today, topic/sof-dev doesn't have this problem.

```
===========================>>
[ 7696.047341] kernel: i915 0000:00:02.0: [drm] *ERROR* Suspending crtc's failed with -22
<<===========================
```

*To reproduce the issue,
TPLG=sof-ehl-rt5660.tplg ~/sof-test/test-case/check-suspend-resume.sh -l 50

**Comment by mengdonglin** [comment]
Libin will check if HDMI audio is impacted by this i915 failure, not high priority.

**Comment by fredoh9** [comment]
should we ignore this i915 error for now?

**Comment by libinyang** [comment]
some update on this bug:
1. I can reproduce this bug. And this is i915 bug, not SOF bug.
2. the i915 has bugs. The GUI will hang when logging in. 
3. Most times, there is no sound w/ or wo suspend/resume. aplay can run successfully on HDMI, but there is no sound.
4. Occasionally, if the screen is flashed automatically (It just flashed unaccountably), the sound is output from the HDMI.

So I think we can ignore this bug as it is not related to SOF audio driver. The i915 is buggy on EHL platform.

**Comment by marc-hb** [comment]
@kv2019i can we safely ignore this one? Only on EHL or on any platform? Only for now or forever?

**Comment by kv2019i** [comment]
@marc-hb and others -- if we can repeat the problem (as seems to be the case), the proper thing would be to file to i915 bugzilla:
https://gitlab.freedesktop.org/drm/intel/-/issues/

Before filing, one should test with DRM-intel tree (drm-intel-next branch at git://anongit.freedesktop.org/drm-intel) and see if bug reproduces (you'll be asked to do this in any case).

Then we can link to i915 here and keep this for monitoring.

**Comment by fredoh9** [comment]
@kv2019i for drm-intel tree, what kernel config we should use? same as ours, nothing is to be specified for any drm for the build?

**Comment by marc-hb** [comment]
@kv2019i  my question was too vague sorry, let me rephrase it. We should of course report this bug on one hand. On the other hand, for our own, "selfish" and pure audio validation purposes can we add a sof-test filter that ignores this specific message and report audio tests as PASSing when there is no other error?

In other words, can this error have any sort of audio impact outside HDMI?

**Comment by fredoh9** [comment]
for some reason, i can't access freedesktop git( git://anongit.freedesktop.org/drm-intel). I found a github mirror instead. I will try with drm-intel-next branch in this afternoon.

https://github.com/freedesktop/drm-intel

**Comment by fredoh9** [comment]
I tried with drm-intel-next, SHA1=989634fb49ad, And SOF FW SHA1=041c2ef6d.
I couldn't reproduce this problem. I'm trying that again but I remember it was easy to reproduce the problem.
Most likely this is fixed in latest kernel upstream then.

**Comment by marc-hb** [comment]
Reproduced again in nightly run number 2935

```
Start Time: 2021-03-30 21:19:46 UTC
End Time: 2021-03-31 02:00:01 UTC
Kernel Branch: topic/sof-dev
Kernel Commit:  e06bad68
SOF Commit:  7c08aabbe730
```

**Comment by marc-hb** [comment]
@kv2019i  ping?
> On the other hand, for our own, "selfish" and pure audio validation purposes can we add a sof-test filter that ignores this specific message and report audio tests as PASSing when there is no other error?
> In other words, can this error have any sort of audio impact outside HDMI?

**Comment by fredoh9** [comment]
We should ignore this error, this only happens with suspend/resume. We can wait @kv2019i confirmation, but I don't have any concern on audio impact.

But more interesting thing is @libinyang was able to reproduce same problem with drm-intel-next branch whereas I wasn't. Libin please update your findings.

**Comment by kv2019i** [comment]
@fredoh9 @marc-hb I'm ok to ignore this if this no practical impact to operation of SOF CI. The follow-up to i915 can be done separately then.

**Comment by plbossart** [comment]
@fredoh9 @marc-hb @kv2019i can this be closed?

**Comment by marc-hb** [comment]
@XiaoyunWu6666 I understand you haven't see this recently? If not please close.

It may have been fixed upstream.

**Comment by kv2019i** [comment]
Let's close this. Not reproduced with upstream -> https://github.com/thesofproject/linux/issues/2778#issuecomment-806358349
